test/e2e/network/service.go

```diff
@@ -3283,42 +3282,72 @@ var _ = SIGDescribe("ESIPP [Slow]", func() {
 			framework.Failf("Service HealthCheck NodePort still present")
 		}
 
-		endpointNodeMap, err := jig.GetEndpointNodes()
+		config := e2enetwork.NewNetworkingTestConfig(f, false, false)
+
```

Comment:
And second thing: as part of NewNetworkingTestConfig above, what happens we also require external IP address:
https://github.com/kubernetes/kubernetes/blob/72fb952dd6cf54f56925853005198957aaf1584d/test/e2e/framework/network/utils.go#L649

@mm4tt - I think this will work, because we have public heapster node, right? Or should it also be changed?

Reply:
I am not pretty sure, but from these lines https://github.com/kubernetes/kubernetes/blob/72fb952dd6cf54f56925853005198957aaf1584d/test/e2e/framework/network/utils.go#L671-L674, it looks like `config.ExternalAddr` is only used to determine `config.NodeIP` and `config.NodeIP` would be set to an internal IP if `config.ExternalAddr` is empty.